### PR TITLE
COLLECT-415 - Adding slicing when uploading lots of form instances

### PR DESCRIPTION
**Pull request by nribeka**

### CODE REVIEW

**Comment by lognaturel** [comment]
@nribeka @yanokwa @nap2000  Should I wait to review this until you have a test in place?

**Comment by nap2000** [comment]
I think it can be reviewed without a complete stress test with over 999.  However the logic should be tested by adjusting the limit and showing it works when there are more  instances than can be sent in one go. Boundary condition testing would be good too.  Ie set the limit to 3 then test with 2,3 and 4 instances.

**Comment by yanokwa** [comment]
I tested this with 1500 small submissions and it works great! It does 999 submissions first, then sends the remaining 501. I can also confirm that boundary testing for small numbers works.

It should be ready to merge pending code review from @lognaturel.

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceUploaderTask.java [hunk 1]
```diff
@@ -585,6 +588,25 @@ protected Outcome doInBackground(Long... values) {
             }
         }
 
+        return true;
+    }
+
+    // TODO: This method is like 350 lines long, down from 400.
```
Comment (yanokwa) [inline collect_app/src/main/java/org/odk/collect/android/tasks/InstanceUploaderTask.java]: Please move these comments.

Comment (nribeka) [reply collect_app/src/main/java/org/odk/collect/android/tasks/InstanceUploaderTask.java]: Gone.

**Comment by lognaturel** [comment]
Congrats on your merged PR, @nribeka. 🎉🎉
